subsys/bluetooth/common/rpa.c

```diff
@@ -14,20 +14,41 @@
 #include <stddef.h>
 #include <errno.h>
 #include <string.h>
-#include <sys/atomic.h>
-#include <sys/util.h>
-#include <sys/byteorder.h>
-#include <debug/stack.h>
-
-#include <tinycrypt/constants.h>
-#include <tinycrypt/aes.h>
-#include <tinycrypt/utils.h>
-#include <tinycrypt/cmac_mode.h>
 
 #define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_RPA)
 #define LOG_MODULE_NAME bt_rpa
 #include "common/log.h"
 
+#include <bluetooth/crypto.h>
+
+#if defined(CONFIG_BT_CTLR) && defined(CONFIG_BT_HOST_CRYPTO)
+#include "../controller/util/util.h"
+#include "../controller/hal/ecb.h"
+#endif /* defined(CONFIG_BT_CTLR) && defined(CONFIG_BT_HOST_CRYPTO) */
+
+static inline int _rand(void *buf, size_t len)
```

Comment:
You can probably drop the `inline` here. Unless you disable inlining with some explicit compiler option any functions that are this short and have a single caller will almost certainly get implicitly inlined. Also, I think there's a general avoidance of underscore-prefixed symbols in Zephyr, so I'd drop it. You could either just call this `rand`, or (perhaps better since it's a common name) put an _internal suffix on it.

Reply:
Ok. using internal_rand then. Since rand is defined by stdlib.h